### PR TITLE
ORC-492: Avoid potential `ArrayIndexOutOfBoundsException` when getting WriterVersion

### PR DESCRIPTION
**Pull request by guiyanakuang**
### What changes were proposed in this pull request?

This pr is aimed at avoiding potential ArrayIndexOutOfBoundsException possible when getting WriterVersion.

### Why are the changes needed?

Fix bug.
Return the Future version when val is equal to versions.length to avoid throwing ArrayIndexOutOfBoundsException on line 248.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

File: java/core/src/java/org/apache/orc/OrcFile.java [hunk 1]
```diff
@@ -242,7 +242,7 @@ public static WriterVersion from(WriterImplementation writer, int val) {
             val + " for writer " + writer);
       }
       WriterVersion[] versions = values[writer.id];
-      if (val < 0 || versions.length < val) {
+      if (val < 0 || val >= versions.length) {
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/OrcFile.java:245]: If you don't mind, could you keep the previous style like the following?
```java
- if (val < 0 || val >= versions.length) {
- if (val < 0 || versions.length <= val) {
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (with one minor style comment)

File: java/core/src/java/org/apache/orc/OrcFile.java (hunk 1 above)
Comment (guiyanakuang) [reply java/core/src/java/org/apache/orc/OrcFile.java:245]: I adjusted the order on purpose, comparing variables on the same side looks better to understand, that's how I feel.
Of course I don't mind changing it back, but I'd like to mention your opinion.

Comment (guiyanakuang) [reply java/core/src/java/org/apache/orc/OrcFile.java]: 6adff23, keep the previous style.  : ）

**Comment by dongjoon-hyun** [comment]
Thank you for updating. Merged to main.

**Comment by dongjoon-hyun** [comment]
I backported this to branch-1.7.

**Pull request merged**